### PR TITLE
📜Terms of service UI changes

### LINKED ISSUE #12122: Change link for privacy policy
Users can access our privacy policy 
- in 'About/Privacy Policy' 
- on the first onboarding screen (see below)

IMG_CBD107FB729F-1

The current link is https://www.iubenda.com/privacy-policy/45710059

This link needs to be changed to our new version of Privacy Policy https://status.im/privacy-policy/

### LINKED ISSUE #12206: New user with new install - require user ToS opt in
When a user downloads Status for the first time, he needs to approve terms of services before being able to proceed with using the app. 

**Design** 
https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=9665%3A33134

**Notes** 
- the checkbox is unchecked by default, the user must check it before being able to continue 
- Terms of Services link is https://status.im/terms-of-service/ and should be displayed in a webview 
- Privacy policy link is https://status.im/privacy-policy/ and should be displayed in a webview

### LINKED ISSUE #12207: Existing user upgrading Status app - require user ToS opt-in
When a user has Status already installed, and upgrades his app, he needs to approve Terms of Services. 

Note: we assume we will upgrade our terms of services concurrently to new app versions. We don't want to setup a push notification service that pushed to user that a newt Term of Services is available. 

The following screen is shown right after install: 
https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=9417%3A32595

image

Links:
- The Terms of Services link is https://status.im/terms-of-service/ and should be displayed in a web view
- The Principles link is https://our.status.im/our-principles/  and should be shown in a webview

To be done: 
- exact copy to be reviewed by @carlyehoward - DONE 
- behaviour of the decline button to be specified @errorists @hesterbruikman - DONE (there is no decline button anymore)

### LINKED ISSUE #12208: Add ToS to Settings
As a user I should be able to check Terms of Services in the setting section of Status

They should be open in a webview

https://www.figma.com/file/TNCyHKtR3sx5EL6YznFWUa4O/Profile?node-id=6537%3A0

### PR DESCRIPTION
**Pull request by shivekkhurana**
fixes #12122 
fixes #12208 
fixes #12206
fixes #12207 

### Summary
- Link to Status website for for privacy policy and terms of service
- Terms of service in about app sections
- Web view instead of browser redirect for ToS and privacy policy at all places
- Opt-in checkbox for new users on onboarding screen
- Opt-in screen for existing users

### Testing notes
Might break onboarding because the user needs to accept ToS (check checkbox). Until checkbox is checked, the "Get started" button is disabled

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
- Onboarding

### Steps to test

ToS opt-in for new users
- Delete all accounts
- On the carousel screen, you will see a checkbox
- The "Get started" button should be disabled until the checkbox is checked
- ToS and privacy policy should open web views linking to Status website

ToS in Settings
- Go to profile > About app
- You should see an item of Terms of service
- Clicking that should open a web view to Status website

ToS opt-in for existing users
- If the app already has some multi accounts and the welcome opt-in was never displayed, the user will be redirected to an opt-in screen (#12207). 
- This screen should not be visible:
  - For new downloads 
  - For users who opted-in but deleted all multi accounts

- This screen should be visible:
  - For users who have one or more multiaccounts but never opted-in during onboarding (at carousel screen)

### CODE REVIEW

File: src/status_im/utils/config.cljs [hunk 1]
```diff
@@ -151,3 +151,7 @@
     "giphy.com"
     "gph.is"
     "media.giphy.com"})
+
+(def privacy-policy-link "https://status.im/privacy-policy/")
```
Comment (flexsurfer) [inline src/status_im/utils/config.cljs]: probably better move to constants, and also add ^const

File: src/status_im/ui/screens/terms_of_service/views.cljs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+   [react/view {:style {:padding-right (:xx-large spacing/spacing)}}
+    (if (keyword? label)
+      [react/text (i18n/label label)]
+      label)]])
+
+(defn force-accept-tos []
+  [react/scroll-view
+   [react/view {:style (merge {:align-items :center}
+                              (:x-large spacing/padding-horizontal))}
+    [react/image {:source (js/require "../resources/images/tokens/mainnet/SNT.png")
```
Comment (flexsurfer) [inline src/status_im/ui/screens/terms_of_service/views.cljs]: there is also `[react/image {:source      (resources/get-image :status-logo)`

**Review by flexsurfer — APPROVED** [review]
great job!

**Comment by Serhy** [comment]
@shivekkhurana could you rebase against latest develop please? (RNN update merged recently)

**Comment by Serhy** [comment]
Thanks @shivekkhurana 

All looks good to me, apart two points:
**Issue 1** I don't see ToS screen for users who upgraded app from release build. I assume it should be visible right after opening Status after upgrade, - but nothing is displayed for me.
Below is the 'manual' build made of this PR branch in order to upgrade release/1.14 (installed from PlayStore)
**Reproduction**:
1) Install Status from PlayStore 
2) Create an account (Version should be `1.14.0 (20210607113624)`)
3) Upgrade Status on to this build http://status-im-builds.ams3.digitaloceanspaces.com/StatusIm-210624-113852-741e51-manual-universal.apk
4) Open Status
**Actual result:** No ToS opt-in screen for existing users displayed

**Issue 2** If user creates an account (ToS must be checked) but then in the same app session user deletes this account, - redirect to carousel view happens, but ToS checkbox is already checked. And so proceeding with creation new/recover of an account does not require user be 'Agree of ToS'. I'm not sure whether it is the same in legal terms to that if "we have checked this checkbox by default when installing Status for the first time", but better to be safe an have always it unchecked once this view is opened. 
(It's all fine in a case Status reopened at least once prior Deleting an account)
**Reproduction**
1) Install Status 
2) Check ToS and create new account
3) Once you are on Chats main view -> delete account by Profile -> Privacy&Security -> Delete account
**Actual result:** Navigation to carousel view with ToS checkbox checked by default

https://user-images.githubusercontent.com/8749671/123267200-6a4e4800-d505-11eb-857b-696350095bce.mp4

**Comment by Serhy** [comment]
Issue 1 and Issue 2 above fixed and all fine: check box must be checked for new builds in order to proceed with creating new account, and if to upgrade an app there is a ToS banner appears for existing accounts after login. 
However with upgrading there is one small issue: after Status upgrade there is an ability to create new account and proceed using it without agreement with ToS since it's not shown. Not shown until the first re-login. We don't check for flag when navigation from `Enable notifications` view to `Chats`

**Reproduction**:
1) Install Status release build from PlayStore (1.14 version)
2) Create new account in 1.14 build
3) Upgrade on this PR build (manual build is from `01bdbc9` is http://status-im-builds.ams3.digitaloceanspaces.com/StatusIm-210629-083334-01bdbc-manual-universal.apk
4) After upgrade, - create second account in Status till you reach `Chats` view
**Actual result:** after creation when you land on Chats main view - there is no ToS banner with Accept and Continue. BUT if I restart Status and login to any account - ToS banner appears fine as expected.
**Expected result:** ToS banner with Accept and continue appears right away on Chats view in the second account created.

**Comment by Serhy** [comment]
Tested on iOS14 and Android 11, looks good to me now!
- on the very beginning of new fresh build there is a ToS checkbox which must be checked prior proceed for new account/recover creation. If to delete all accounts - user need to check ToS again
- after upgrade, - (from Status 1.14 to this PR build) when login into any existing account there is a ToS opt-in banner appears. If close an app without accepting it appears again.

Thanks, @shivekkhurana for implementing and fixing some bugs above which were here.

**Pull request merged**